source/core/owt_base/selector/VideoQualitySwitch.h

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) <2021> Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OWT_BASE_SELECTOR_VIDEO_QUALITY_SWITCH_H
+#define OWT_BASE_SELECTOR_VIDEO_QUALITY_SWITCH_H
+
+#include "MediaFramePipeline.h"
+#include <logger.h>
+
+#include <deque>
+#include <memory>
+#include <vector>
+
+namespace owt_base {
+
+class VideoQualitySwitch : public FrameSource,
+                           public FrameDestination {
+    DECLARE_LOGGER();
+public:
+    VideoQualitySwitch(std::vector<FrameSource*> sources);
+    ~VideoQualitySwitch();
+
+    // Implements FrameDestination
+    void onFrame(const Frame&) override;
+    void onMetaData(const MetaData&) override;
+
+    // Implements FrameDestination
+    void onFeedback(const owt_base::FeedbackMsg& msg) override;
+
+    void setTargetBitrate(uint32_t targetBps);
+
+    class BitrateCounter : public FrameDestination {
```

Comment:
Please use rtc_base/RateStatistics class from WebRTC for this purpose instead of implementing your own.

Reply:
It's not in webrtc addon, I don't want to link it with webrtc lib.